### PR TITLE
[SDFAB-1026] Add reconciliation cycle for QER meters

### PR DESCRIPTION
**Pull request by daniele-moro**

### CODE REVIEW

File: app/app/src/main/java/org/omecproject/up4/impl/Up4DeviceManager.java [hunk 1]
```diff
@@ -1359,5 +1376,62 @@ private void checkStateAndReconcile() throws UpfProgrammableException {
...
+                // Add missing: Rule is in the leader but not in the follower
+
+                Set<Meter> unexpectedMeters = followerMeters.stream()
+                        .filter(fr -> leaderMeters.stream().noneMatch(lr -> lr.equals(fr)))
+                        .collect(Collectors.toSet());
+                followerMeters.removeAll(unexpectedMeters);
+
+                Set<Meter> staleMeters = leaderMeters.stream()
+                        .filter(lr -> followerMeters.stream().noneMatch(fr -> fr.equals(lr)
+                                && fr.bands().containsAll(lr.bands())))
```
Comment (pierventre) [inline app/app/src/main/java/org/omecproject/up4/impl/Up4DeviceManager.java:1416]: are we doing an assumption on the bands size ? Maybe worth to drop a comment as reminder

**Review by pierventre — APPROVED** [review]

File: app/app/src/main/java/org/omecproject/up4/impl/Up4DeviceManager.java [hunk 2]
```diff
@@ -1359,5 +1376,62 @@ private void checkStateAndReconcile() throws UpfProgrammableException {
...
+                Set<Meter> followerMeters =
+                        meterService.getMeters(deviceId).stream()
+                                .filter(upfProg::fromThisUpf)
+                                .filter(m -> m.state() == MeterState.PENDING_ADD || m.state() == MeterState.ADDED)
+                                .map(m -> copyMeterForDevice(m, leaderUpfDevice))
+                                .collect(Collectors.toSet());
+
+                // Remove unexpected: Meter is in the follower but not in the leader
+                // Update stale: Meter is both on follower and leader but bands are different
+                // Add missing: Rule is in the leader but not in the follower
```
Comment (Wailoks) [inline app/app/src/main/java/org/omecproject/up4/impl/Up4DeviceManager.java:1407]: Meter

File: app/app/src/main/java/org/omecproject/up4/impl/Up4DeviceManager.java (hunk 1 above)
Comment (daniele-moro) [reply app/app/src/main/java/org/omecproject/up4/impl/Up4DeviceManager.java]: added a check on the number of bands, this should solve the issue

**Review by pierventre — APPROVED** [review]
We can merge it when you feel ready

**Pull request merged**